DOCS.md

```diff
@@ -66,6 +66,30 @@ You can reference the n-th matched group with `'\\n'` (`'\\0'` refers to the who
 
 To use the backslash character (`\`) just escape it like so: `'\\\\'` (double escape is needed because of JSON already using `\` for escaping).
 
+### Regular expressions with a function
+
+If you need even more power over the aliased path, you can pass a function in the alias configuration:
+
+```js
+module.exports = {
+  plugins: [
+    ["module-resolver", {
+      alias: {
+        "^@namespace/foo-(.+)": ([, name]) => path.join('packages', name)
+      }
+    }]
+  ]
+}
+```
+
+Using the config from this example `'@namespace/foo-bar'` will become `'packages/bar'` or `'packages\\bar'` depending on the OS.
+
+The only argument is the result of calling `RegExp.prototype.exec` on the matched path. It's an array with the matched string and all matched groups.
+
+Because the function is only called when there is a match, the argument can never be `null`.
+
+*Important: a function can only be used with regular expressions.*
```

Comment:
Aren't all aliases converted into regex anyway?

Reply:
They are, but the value for the "basic" alias is made into a substitution string like so:
```js
getAliasPair(`^${key}(/.*|)$`, `${alias[key]}\\1`);
```

I could add some more logic there to always detect functions.

Reply:
Yup I think being consistent here would be less surprising and the note in the DOCS wouldn't be necessary. Win-win!

Reply:
Done.